spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe 'GET posts', type: :request do
+  it 'works! returns correct status code' do
+    get user_posts_path(user_id: 1)
+    expect(response).to have_http_status(200)
+  end
+
+  it 'works!  renders index view correctly' do
+    get user_posts_path(user_id: 1)
+    expect(response).to render_template(:index)
+  end
+
+  it 'works! reponse body includes <h1>This is a list of posts<h1>' do
+    get user_posts_path(user_id: 1)
+    expect(response.body).to include('<h1>This is a list of posts<h1>')
+  end
+end
+
+RSpec.describe 'GET posts/:id', type: :request do
+  it 'works! returns correct status code' do
+    get user_post_path(1, id: 1)
+    expect(response).to have_http_status(200)
+  end
+
+  it 'works!  renders show view correctly' do
+    get user_post_path(1, id: 1)
+    expect(response).to render_template(:show)
+  end
+
+  it 'works! reponse body includes <h1>this is a single post </h1> ' do
+    get user_post_path(1, id: 1)
+    expect(response.body).to include('<h1>this is a single post </h1>')
+  end
+end
```

Comment:
- [ ] Going back to the previous review concern. What was meant is that you should only have one **_`Rspec.describe`_** block per test file. What you should have done is take the test examples you have in the second **_`Rspec.describe`_** and move them first one. 

NOTE: It is allowed to have other **_describe_** blocks inside the main **_`Rspec.describe`_** block if separating contexts was the concern you were addressing.